src/generic_app.ts

```diff
@@ -259,28 +390,89 @@ export class PolkadotGenericApp extends BaseApp {
   }
   /**
+   * @deprecated Use signRawEcdsa or signRawEd25519 instead. This method will be removed in a future version.
    * Signs a raw transaction blob.
    * @param path - The BIP44 path.
    * @param txBlob - The transaction blob.
+   * @param scheme - The scheme to use for the signing. Default is ED25519.
    * @throws {ResponseError} If the response from the device indicates an error.
    * @returns The response containing the signature and status.
    */
-  async signRaw(path: BIP32Path, txBlob: TransactionBlob) {
-    return await this.signImpl(path, this.INS.SIGN_RAW, txBlob)
+  async signRaw(path: BIP32Path, txBlob: TransactionBlob, scheme = SCHEME.ED25519) {
+    if (scheme != SCHEME.ECDSA && scheme != SCHEME.ED25519) {
+      throw new ResponseError(LedgerError.ConditionsOfUseNotSatisfied, `Unexpected scheme ${scheme}. Needs to be ECDSA (2) or ED25519 (0)`)
+    }
+    if (scheme === SCHEME.ECDSA) {
+      return await this.signRawEcdsa(path, txBlob)
+    }
+    return await this.signRawEd25519(path, txBlob)
   }
 
   /**
+   * Signs a raw transaction blob using the ED25519 scheme.
+   * @param path - The BIP44 path.
+   * @param txBlob - The transaction blob.
+   * @throws {ResponseError} If the response from the device indicates an error.
+   * @returns The response containing the signature and status.
+   */
+  async signRawEd25519(path: BIP32Path, txBlob: TransactionBlob) {
+    return await this.signImplEd25519(path, this.INS.SIGN_RAW, txBlob)
+  }
+
+  /**
+   * Signs a raw transaction blob using the ECDSA scheme.
+   * @param path - The BIP44 path.
+   * @param txBlob - The transaction blob.
+   * @throws {ResponseError} If the response from the device indicates an error.
+   * @returns The response containing the signature and status.
+   */
+  async signRawEcdsa(path: BIP32Path, txBlob: TransactionBlob) {
+    return await this.signImplEcdsa(path, this.INS.SIGN_RAW, txBlob)
+  }
+
+  /**
+   * @deprecated Use signWithMetadataEd25519 or signWithMetadataEcdsa instead. This method will be removed in a future version.
    * [Expert-only Method] Signs a transaction blob with provided metadata (this could be used also with a migration app)
    * @param path - The BIP44 path.
    * @param txBlob - The transaction blob.
    * @param txMetadata - The transaction metadata.
    * @throws {ResponseError} If the response from the device indicates an error.
    * @returns The response containing the signature and status.
    */
-  async signWithMetadata(path: BIP32Path, txBlob: TransactionBlob, txMetadata: TransactionMetadataBlob) {
-    return await this.signImpl(path, this.INS.SIGN, txBlob, txMetadata)
+  async signWithMetadata(path: BIP32Path, txBlob: TransactionBlob, txMetadata: TransactionMetadataBlob, scheme: SCHEME) {
+    if (scheme != SCHEME.ECDSA && scheme != SCHEME.ED25519) {
+      throw new ResponseError(LedgerError.ConditionsOfUseNotSatisfied, `Unexpected scheme ${scheme}. Needs to be ECDSA (2) or ED25519 (0)`)
```

Comment:
Should this be ECDSA = 1 and ED25519 = 0?

Reply:
I was keeping the enum we had for polkadot where 1 was for sr255

Reply:
Ok, amazing! I like it then!